File: genetic-api/src/main/java/genetic/api/individual/Promoted.java
package genetic.api.individual;


import genetic.api.elitism.EliteAcknowledged;
import genetic.api.elitism.Promoter;
import genetic.api.individual.impl.Elite;

/**
 * @author piotr.larysz
 */
public interface Promoted<T1 extends Comparable<T1>, T2>  {

    <M extends Promoter & EliteAcknowledged> Elite<T1, T2> promote(M promotor);
}

File: genetic-api/src/main/java/genetic/api/individual/Promoted.kt
package genetic.api.individual

import genetic.api.elitism.EliteAcknowledged
import genetic.api.elitism.Promoter
import genetic.api.individual.impl.Elite

interface Promoted<T> {

  fun <M> promote(promoter: M): Elite<T> where M : Promoter, M : EliteAcknowledged
}